services/checks/funcs.go

```diff
@@ -80,3 +85,79 @@ func formatVersionNum(args ...interface{}) (interface{}, error) {
 	}
 	return p.String(), nil
 }
+
+// GetAdditionalContext returns additional functions to be used in check scripts.
+func GetAdditionalContext() map[string]starlark.GoFunc {
+	return map[string]starlark.GoFunc{
+		"ip_is_private":      ipIsPrivate,
+		"parse_version":      parseVersion,
+		"format_version_num": formatVersionNum,
+	}
+}
+
+// ipIsPrivate accepts a single string argument (IP address) and
+// returns true for a private address, otherwise false.
+func ipIsPrivate(args ...interface{}) (interface{}, error) {
+	if l := len(args); l != 1 {
+		return nil, errors.Errorf("expected 1 argument, got %d", l)
+	}
+
+	ip, ok := args[0].(string)
+	if !ok {
+		return nil, errors.Errorf("expected string argument, got %[1]T (%[1]v)", args[0])
+	}
+
+	ipAddress := net.ParseIP(ip)
+	if ipAddress == nil {
+		// check if string was in CIDR notation
+		_, net, err := net.ParseCIDR(ip)
+		if err != nil {
+			return nil, errors.Errorf("invalid ip address: %s", ip)
+		}
+		for _, network := range privateNetworks {
+			if reflect.DeepEqual(net, network) {
+				return true, nil
+			}
+		}
+		return false, nil
+	}
+
+	for _, network := range privateNetworks {
+		if network.Contains(ipAddress) {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func init() {
+	// full list of reserved network addresses https://en.wikipedia.org/wiki/Reserved_IP_addresses
+	privateAddressBlocks := []string{
+		// private blocks, see https://tools.ietf.org/html/rfc1918
+		"10.0.0.0/8",
+		"172.16.0.0/12",
+		"192.168.0.0/16",
+		// shared address space range, see https://tools.ietf.org/html/rfc6598#section-7
+		"100.64.0.0/10",
+		// IETF protocol assignments range, see https://tools.ietf.org/html/rfc5735#section-4
+		"192.0.0.0/24",
+		// see https://tools.ietf.org/html/rfc5735#section-4
+		"198.18.0.0/15",
+		// link-local block, see https://tools.ietf.org/html/rfc3927
+		"169.254.0.0/16",
+		// loop-back block, see https://tools.ietf.org/html/rfc5735
+		"127.0.0.0/8",
+
+		// IPv6 Unique Local Address prefix analogous to IPv4 private networks, see https://tools.ietf.org/html/rfc4193#section-3
+		"fc00::/7",
+		// IPv6 link-local prefix, see https://tools.ietf.org/html/rfc4291#section-2.4
+		"fe80::/10",
+		// IPv6 loop-block prefix, see https://tools.ietf.org/html/rfc4291#section-2.4
+		"::1/128",
+	}
+
+	for _, b := range privateAddressBlocks {
+		_, network, _ := net.ParseCIDR(b)
```

Comment:
ignoring `err` here since all the cidr provided by us are valid and won't return any error and also because `init` can't handle errors

Reply:
> provided by us are valid

How do we know that?

> also because init can't handle errors

panic(err) :)

Reply:
added panic, although I don't like the idea of having a package that can panic